### PR TITLE
docs: RangePicker type and default value in doc

### LINKED ISSUE #30718: Incorrect type for separator RangePicker
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/date-picker/#RangePicker](https://ant.design/components/date-picker/#RangePicker)

### Steps to reproduce
You can see in the doc the type for separator is `string`, but it should be `React.ReactNode` according to the codebase

### What is expected?
 it should be `React.ReactNode` according to the codebase

### What is actually happening?
The doc def is wrong

| Environment | Info |
|---|---|
| antd | 4.16.0 |
| React | ^16.14.0 |
| System | MacOS |
| Browser | Chrome 90 |

### PR DESCRIPTION
**Pull request by LookRain**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/30718

### 💡 Background and solution

The type def and default value for `separator` prop for `RangePicker` is wrong. It should accept `React.ReactNode`, and with a default value `<SwapRightOutlined />`. Currently it's `string` and `~`.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix the documentation's type definition and default value for `separator` prop for `RangePicker`       |
| 🇨🇳 Chinese |     改正文档中`RangePicker`里`separator` prop的type和default value    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
@xrkffgg 上面那个action判断逻辑似乎要改一下。

**Comment by xrkffgg** [comment]
> @xrkffgg 上面那个action判断逻辑似乎要改一下。

已修~ 😘

**Comment by LookRain** [comment]
@afc163 我看这个`Semantic Pull Request `没过，需要修改commit message吗？

**Comment by afc163** [comment]
要。

File: components/date-picker/index.en-US.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ Added in `4.1.0`.
 | format | To set the date format, refer to [moment.js](http://momentjs.com/). When an array is provided, all values are used for parsing and first value is used for formatting | string \| string\[] | `YYYY-MM-DD HH:mm:ss` |  |
 | ranges | The preseted ranges for quick selection | { \[range: string]: [moment](http://momentjs.com/)\[] } \| { \[range: string]: () => [moment](http://momentjs.com/)\[] } | - |  |
 | renderExtraFooter | Render extra footer in panel | () => React.ReactNode | - |  |
-| separator | Set separator between inputs | string | `~` |  |
+| separator | Set separator between inputs | React.ReactNode | &lt;SwapRightOutlined /> |  |
```
Comment (afc163) [inline components/date-picker/index.en-US.md:163]:
```suggestion
| separator | Set separator between inputs | React.ReactNode | `<SwapRightOutlined />` |  |
```

**Pull request merged**